coderd/workspacebuilds.go

```diff
@@ -277,6 +280,22 @@ func (api *API) postWorkspaceBuilds(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// we only want to create audit logs for delete builds right now
+	if action == rbac.ActionDelete {
+		var (
+			auditor           = api.Auditor.Load()
+			aReq, commitAudit = audit.InitRequest[database.Workspace](rw, &audit.RequestParams{
+				Audit:   *auditor,
+				Log:     api.Logger,
+				Request: r,
+				Action:  database.AuditActionDelete,
+			})
+		)
+
+		defer commitAudit()
+		aReq.Old = workspace
+	}
+
```

Comment:
@coadler I'd love to write a test asserting log creation as we discussed, but I am having some difficulty.

In `coderd/workspacebuilds_test.go` in the `TestWorkspaceBuildStatus` test, I attempted to add:

````
require.Len(t, auditor.AuditLogs, 4)
assert.Equal(t, database.AuditActionDelete, auditor.AuditLogs[3].Action)
````

at the very end of the test, after successful deletion by the client; however, `auditor.AuditLogs` is always an empty list. Any idea why?

Reply:
Could you push that code up too? I'll try and take a look!

Reply:
@coadler pushed! Thanks!